pgjdbc/src/test/java/org/postgresql/test/jdbc4/jdbc41/SchemaTest.java

```diff
@@ -217,6 +223,74 @@ public void testSearchPathPreparedStatement() throws SQLException {
     ps.close();
   }
 
+  @Test
+  public void testCurrentSchemaPropertyVisibilityTableDuringFunctionCreation() throws SQLException {
+    Properties properties = new Properties();
+    properties.setProperty(PGProperty.CURRENT_SCHEMA.getName(), "public,schema1,schema2");
+    Connection connection = TestUtil.openDB(properties);
+
+    TestUtil.execute("create table schema1.check_table (test_col text)", connection);
+    TestUtil.execute("insert into schema1.check_table (test_col) values ('test_value')", connection);
+    TestUtil.execute("create or replace function schema2.check_fun () returns text as $$"
+        + " select test_col from check_table"
+        + "$$ language sql immutable", connection);
+    connection.close();
+  }
+
+  @Test
+  public void testCurrentSchemaPropertyNotVisibilityTableDuringFunctionCreation() throws SQLException {
+    Properties properties = new Properties();
+    properties.setProperty(PGProperty.CURRENT_SCHEMA.getName(), "public,schema2");
+    Connection connection = TestUtil.openDB(properties);
+
+    TestUtil.execute("create table schema1.check_table (test_col text)", connection);
+    TestUtil.execute("insert into schema1.check_table (test_col) values ('test_value')", connection);
+    try {
+      TestUtil.execute("create or replace function schema2.check_fun (txt text) returns text as $$"
+          + " select test_col from check_table"
+          + "$$ language sql immutable", connection);
+    } catch (PSQLException e) {
+      String sqlState = e.getSQLState();
+      String message = e.getMessage();
+      assertThat("Incorrect sql error code", sqlState, equalTo(PSQLState.RELATION_DOES_NOT_EXIST.getState()));
+      assertTrue("Missing table name in error message", message.contains("\"check_table\""));
```

Comment:
Same thing: `Missing table name in error message`.

1) It is not clear what is missing
2) It is not clear why it is missing.


Assert message should better be like `"\"check_table\" should be contained in the exception message because ..."`.
Note: the assertion **hides** the original message, thus it makes analysis hard.
So you should either use `assertThat(..., message, contains("..."))` or add `message` to the message parameter.

Reply:
fixed